alf/metrics/metrics.py

```diff
@@ -165,16 +165,12 @@ def _extract_metric_values(self, time_step):
         raise NotImplementedError()
 
     def _initialize(self, example_metric_value):
-        counter = [0]
-
         def _init_buf(val):
             return MetricBuffer(max_len=self._buffer_size, dtype=self._dtype)
 
         def _init_acc(val):
             accumulator = torch.zeros(
                 self._batch_size, dtype=self._dtype, device='cpu')
-            self.register_buffer('_accumulator%d' % counter[0], accumulator)
```

Comment:
why do you remove this?
This is needed so that accumulator can be checkpointed.

Reply:
This is actually from #1083, and apparently this PR has not been rebased to the latest master. 

As suggested by @hnyu , the accumulator is better not check-pointed, so that when a training is resumed from a checkpoint, the accumulating restarts.